internal/hmyapi/apiv1/error.go

```diff
@@ -1,8 +1,35 @@
 package apiv1
 
-import "errors"
+import (
+	"errors"
+
+	"github.com/ethereum/go-ethereum/rpc"
+	"github.com/harmony-one/harmony/shard"
+)
 
 var (
-	// ErrIncorrectChainID is an incorrect chain ID.
-	ErrIncorrectChainID = errors.New("incorrect chain ID")
+	// ErrInvalidLogLevel when invalid log level is provided
+	ErrInvalidLogLevel = errors.New("invalid log level")
+	// ErrIncorrectChainID when ChainID does not match running node
+	ErrIncorrectChainID = errors.New("incorrect chain id")
+	// ErrInvalidChainID when ChainID of signer does not match that of running node
+	ErrInvalidChainID = errors.New("invalid chain id for signer")
+	// ErrNotBeaconShard when rpc is called on not beacon chain node
+	ErrNotBeaconShard = errors.New("cannot call this rpc on non beaconchain node")
+	// ErrRequestedBlockTooHigh when given block is greater than latest block number
+	ErrRequestedBlockTooHigh = errors.New("requested block number greater than current block number")
 )
+
+func (s *PublicBlockChainAPI) isBeaconShard() error {
```

Comment:
don't need these functions here. move it to respective blockchain.go for both v1 and v2.

Reply:
Put these functions here, because they can also apply to the transaction pool RPCs, but since transaction refactor in progress, I didn't want to touch the pool RPCs for now.

Reply:
I would still prefer them in blockchain.go, as they are more blockchain related than simply error. Generally, error.go contains error definitions and pre/post-processing errors, nothing more.